tests/test_apply.py

```diff
@@ -209,106 +188,117 @@ def test_invalid_qubit_state_vector(self, device):
         vector is the wrong size"""
         dev = device(2)
         state = np.array([0, 123.432])
+        wires = [0, 1]
 
         with pytest.raises(ValueError, match=r"State vector must be of length 2\*\*wires"):
-            dev.apply("QubitStateVector", [0, 1], [state])
+            dev.apply([qml.QubitStateVector(state, wires=wires)])
 
-    @pytest.mark.parametrize("name,mat", single_qubit)
-    def test_single_qubit_no_parameters(self, init_state, device, name, mat, tol):
-        """Test PauliX application"""
+    @pytest.mark.parametrize("mat", [U, U2])
+    def test_qubit_unitary(self, init_state, device, mat, tol):
+        """Test that the QubitUnitary operation works fine with the apply
+        method."""
+        N = int(np.log2(len(mat)))
+        dev = device(N)
+        state = init_state(N)
+        wires = list(range(N))
+
+        dev.apply([qml.QubitStateVector(state, wires=wires), qml.QubitUnitary(mat, wires=wires)])
+        dev._samples = dev.generate_samples()
+
+        res = np.fromiter(dev.probability(), dtype=np.float64)
+        expected = np.abs(mat @ state) ** 2
+        assert np.allclose(res, expected, **tol)
+
+    def test_invalid_qubit_unitary(self, device):
+        """Test that an exception is raised if the
+        unitary matrix is the wrong size"""
+        dev = device(2)
+        state = np.array([[0, 123.432], [-0.432, 023.4]])
+
+        with pytest.raises(ValueError, match=r"Unitary matrix must be of shape"):
+            dev.apply([qml.QubitUnitary(state, wires=[0, 1])])
+
+    @pytest.mark.parametrize("operation", single_qubit_operations)
+    def test_single_qubit_operations_no_parameters(self, init_state, device, operation, tol):
+        """Test that single qubit operations that take no parameters work fine
+        with the apply method."""
         dev = device(1)
         state = init_state(1)
+        wires = [0]
+        applied_operation = operation(wires=wires)
 
-        dev.apply("QubitStateVector", [0], [state])
-        dev.apply(name, [0], [])
-        dev._obs_queue = []
-        dev.pre_measure()
+        dev.apply([qml.QubitStateVector(state, wires=wires), applied_operation])
+        dev._samples = dev.generate_samples()
 
-        res = np.fromiter(dev.probability().values(), dtype=np.float64)
-        expected = np.abs(mat @ state) ** 2
+        res = np.fromiter(dev.probability(), dtype=np.float64)
+        expected = np.abs(applied_operation.matrix @ state) ** 2
         assert np.allclose(res, expected, **tol)
 
     @pytest.mark.parametrize("theta", [0.5432, -0.232])
-    @pytest.mark.parametrize("name,func", single_qubit_param)
-    def test_single_qubit_parameters(self, init_state, device, name, func, theta, tol):
-        """Test PauliX application"""
+    @pytest.mark.parametrize("operation", single_qubit_operations_param)
+    def test_single_qubit_operations_parameters(self, init_state, device, operation, theta, tol):
```

Comment:
Is it possible that this test appears twice? And that ``two_qubit_pars`` is missing?

Reply:
Right! This was also something I noticed and had a commit yesterday, could you check if it should be fine now? The idea would be to have one case for each per `class`

Reply:
Great, looks good now!